### PR TITLE
IODEMO: add listen retry for server

### PR DESCRIPTION
**Pull request by evgeny-leksikov**

### CODE REVIEW

**Comment by evgeny-leksikov** [comment]
@yosefe pls take a look

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -628,7 +628,25 @@ class DemoServer : public P2pDemoCommon {
         listen_addr.sin_addr.s_addr = INADDR_ANY;
         listen_addr.sin_port        = htons(opts().port_num);
 
-        listen((const struct sockaddr*)&listen_addr, sizeof(listen_addr));
+        for (long i = 1;;) {
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc]: i -> retry

File: test/apps/iodemo/io_demo.cc [hunk 2]
```diff
@@ -628,7 +628,25 @@ class DemoServer : public P2pDemoCommon {
...
+                break;
+            }
+
+            UcxLog log(LOG_PREFIX);
+            log << "retry " << i;
+            if (opts().retries < std::numeric_limits<long>::max()) {
+                log << "/" << opts().retries;
+            }
+
+            if (++i <= opts().retries) {
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc]: if (retry >= retries) return

File: test/apps/iodemo/io_demo.cc [hunk 3]
```diff
@@ -628,7 +628,25 @@ class DemoServer : public P2pDemoCommon {
...
+                log << " reconnecting in " << opts().retry_interval << " seconds";
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc]: overall log message should be
"restarting listener on 0.0.0.0:2000 in 10 seconds (retry 1/5)"
...
"restarting listener on 0.0.0.0:2000 in 10 seconds (retry 5/5)"

**Review by yosefe — APPROVED** [review]

**Pull request merged**